.github/workflows/build-windows.yml

```diff
@@ -0,0 +1,297 @@
+name: windows build
+
+on:
+  push:
+  pull_request:
+
+env:
+  BACKWARDS_COMPATIBILITY_ARRAYS: OFF
+  TILEDB_S3: OFF
+  TILEDB_AZURE: OFF
+  TILEDB_GCS: OFF
+  TILEDB_SERIALIZATION: OFF
+  TILEDB_STATIC: OFF
+  TILEDB_TOOLS: ON
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-2016, windows-2019]
+        #considering https://stackoverflow.com/questions/65035256/how-to-access-matrix-variables-in-github-actions
+        environ: [azure, s3, serialization]
+        include:
+          - environ: 'azure'
+            TILEDB_AZURE: ON
+            TILEDB_ARROW_TESTS: ON
+            TILEDB_SERIALIZATION: OFF
+            TILEDB_S3: OFF
+          - environ: 's3'
+            TILEDB_S3: ON
+            TILEDB_ARROW_TESTS: ON
+            TILEDB_SERIALIZATION: OFF
+            TILEDB_AZURE: OFF
+          - environ: 'serialization'
+            TILEDB_S3: OFF
+            TILEDB_AZURE: OFF
+            TILEDB_ARROW_TESTS: OFF
+            TILEDB_SERIALIZATION: ON
+    env:
+      TILEDB_HOME: ${{ github.workspace }}
+      TILEDB_GA_IMAGE_NAME: ${{ matrix.os }}
+      TILEDB_S3: ${{ matrix.TILEDB_S3 }}
+      TILEDB_AZURE: ${{ matrix.TILEDB_AZURE }} #azure }}
+      TILEDB_SERIALIZATION: ${{ matrix.TILEDB_SERIALIZATION }} #serialization }}
+      TILEDB_ARROW_TESTS: ${{ matrix.TILEDB_ARROW_TESTS }}
+    steps:
+      - name: 'tiledb env prep'
+        run: |
+          $env:BUILD_BUILDDIRECTORY = $env:GITHUB_WORKSPACE.replace("TileDB\TileDB","tdbbd") # 't'ile'db' 'b'uild 'd'ir
+          $env:BUILD_SOURCESDIRECTORY = $env:GITHUB_WORKSPACE
+
+          Write-Host "GITHUB_WORKSPACE is '$env:GITHUB_WORKSPACE'"
+          Write-Host "BUILD_BUILDDIRECTORY is '$env:BUILD_BUILDDIRECTORY'"
+          Write-Host "BUILD_SOURCESDIRECTORY is '$env:BUILD_SOURCESDIRECTORY'"
+          Write-Host "GITHUB_ENV is $env:GITHUB_ENV"
+
+          mkdir $env:BUILD_BUILDDIRECTORY
+
+          write-host "BUILD_BUILDDIRECTORY=$env:BUILD_BUILDDIRECTORY" *>> $env:GITHUB_ENV
+          write-host "BUILD_SOURCESDIRECTORY=$env:BUILD_SOURCESDIRECTORY" *>> $env:GITHUB_ENV
+      - name: 'Print env'
+        run: |
+          set
+        shell: cmd
+      - name: ARROW python needs
+        shell: bash
+        if: ${{ matrix.TILEDB_ARROW_TESTS == 'ON' }}
+        run: |
+          set -e pipefail
+          if [[ "$TILEDB_ARROW_TESTS" == "ON" ]]; then
+            pip install pyarrow pybind11 numpy
+          fi
+      - name: Prepare git
+        run: git config --global core.autocrlf false
+      - uses: actions/checkout@v2
+      - name: core tiledb windows build 
+        run: |
+          $ErrorView = "NormalView" 
+          
+          Write-Host "GITHUB_WORKSPACE is '$env:GITHUB_WORKSPACE'"

```

Comment:
Probably don't need to print these in every step now?

Reply:
removed from secondary steps